src/panels/lovelace/views/hui-sections-view.ts

```diff
@@ -118,17 +148,30 @@ export class SectionsView extends LitElement implements LovelaceViewElement {
         <div
           class="container"
           style=${styleMap({
-            "--max-columns-count": maxColumnsCount,
-            "--total-count": totalCount,
+            "--total-section-count": totalSectionCount,
+            "--max-column-count": maxColumnCount,
           })}
         >
           ${repeat(
             sections,
             (section) => this._getSectionKey(section),
             (section, idx) => {
+              const sectionConfig = this._config?.sections?.[idx];
+              const columnSpan = Math.min(
+                sectionConfig?.column_span || 1,
+                maxColumnCount
+              );
+
               (section as any).itemPath = [idx];
+              (section as any).columnSpan = columnSpan;
+
               return html`
-                <div class="section">
+                <div
+                  class="section"
+                  style=${styleMap({
+                    "--column-span": columnSpan,
```

Comment:
we are settings this variables here and in `hui-grid-section`, can we remove it from `hui-grid-section` as it would be inherited from here?

Reply:
I was exposing the value in javascript so custom section or future section can use it differently. Put I agree, we don't need to overcomplicate it for now. Keep it simple !